src/test/java/BaseTest.java

```diff
@@ -1,11 +1,135 @@
 import org.testng.annotations.BeforeSuite;
 
+import java.time.Duration;
+import java.util.UUID;
+
 public class BaseTest {
+    public static WebDriver driver = null;
+    public static String url = "https://qa.koel.app/";
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+            public void launchBrowser() {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+    }
+
+    @AfterMethod
+    public void closeBrowser(){
+        driver.quit();
+    }
+
+    protected static void clickSubmit() {
+        WebElement submitLogin = driver.findElement(By.cssSelector("button[type='submit']"));
+        submitLogin.click();
+    }
+
+    protected static void enterPassword(String password) {
+        WebElement passwordInput = driver.findElement(By.cssSelector("[type='password']"));
+        passwordInput.click();
+        passwordInput.clear();
+        passwordInput.sendKeys(password);
+    }
+
+    protected static void enterEmail(String email) {
+        WebElement emailInput = driver.findElement(By.cssSelector("[type='email']"));
+        emailInput.click();
+        emailInput.clear();
+        emailInput.sendKeys(email);
+
+    }
+
+    protected static void openLoginURL() {
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+    }
+
+    protected static void clickAvatarIcon(){
+        WebElement avatarIcon = driver.findElement(By.cssSelector("img.avatar"));
+        avatarIcon.click();
+    }
+
+    protected static String generateRandomName(){
+        return UUID.randomUUID().toString().replace("-","");
+    }
+
+    protected static void providePassword(String password){
+        WebElement currentPassword = driver.findElement(By.cssSelector("[name='current_password']"));
+        currentPassword.clear();
+        currentPassword.sendKeys(password);
+    }
+
+    protected static void provideProfileName(String name){
+        WebElement profileName = driver.findElement(By.cssSelector("[name='name']"));
+        profileName.clear();
+        profileName.sendKeys(name);
+    }
+
+    protected static void clickSaveButton(){
+        WebElement saveButton = driver.findElement(By.cssSelector("button.btn-submit"));
+
+    }
+
+
+    //HOMEWORK 17 BELOW
+    //
+    //
+    //
+    //
+    //
+
+    protected void verifySuccessMessageText() {
+        WebElement successNewPlaylist = driver.findElement(By.xpath(
+                "//div[contains(text(), 'new playlist17')]"));
+        Assert.assertTrue(successNewPlaylist.isDisplayed());
+    }
+
+    protected void selectSavePlaylist() {
+        WebElement savePlaylistButton = driver.findElement(By.xpath("//*[@id='songResultsWrapper']/header/div[3]/div/section[2]/form/button"));
```

Comment:
at this selector savePlaylistButton, it's too long. You can find the selector based on the specific attributes of the element (as type, title, ect)
//HINT:
`//*[@id='songResultsWrapper']//button[@title='Save']`

Reply:
@duykhanhrc I appreciate the feedback! i just tried using the selector you provided and i don't think it works, unless i'm doing something wrong. 
I also tried using the same order of actions as you mentioned in the comment above when i was doing this homework, but had no luck, and only then i used the devtools-generated selectors.